### PR TITLE
Adding Wizard Template

### PR DESCRIPTION
**Pull request by anjalia**
**Updated to include {{ ADS EDIT }} where relevant
- Adds wizard/dialog templates, including generic wizard template, generic dialog template, file-saving wizard, and database-querying wizard
- Updates index.js to show these template options
- Adds tests for these templates

### CODE REVIEW

File: generators/app/templates/ext-wizard/src/wizards/db-ops/wizard/pages/page1.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+        let connections : string[] = Array.from(this.serverOptions.keys());
+        let firstConnection = this.serverOptions.get(connections[0]);
+        if (firstConnection) {
+            this.model.server = firstConnection;
+            this.serverDropdown.updateProperties({
+                values: connections
+            });
+        }
+		return firstConnection !== undefined;
```
Comment (llali) [inline generators/app/templates/ext-wizard/src/wizards/db-ops/wizard/pages/page1.ts:103]: please either use space or tabs.

File: generators/app/templates/ext-wizard/src/wizards/db-ops/wizard/api/basePage.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	public readonly wizardPage: azdata.window.WizardPage;
+	protected readonly model: ConnectionModel;
+	protected readonly view: azdata.ModelView;
+	protected readonly width: number; // width of the page
+
+	constructor(wizardPage: azdata.window.WizardPage, model: ConnectionModel, view: azdata.ModelView, width : number) {
+		this.wizardPage = wizardPage;
+		this.model = model;
+		this.view = view;
+		this.width = width;
```
Comment (llali) [inline generators/app/templates/ext-wizard/src/wizards/db-ops/wizard/api/basePage.ts:17]: looks like "space" is used in other file so place use space here as well

File: generators/app/templates/ext-wizard/src/wizards/db-ops/wizard/api/basePage.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import * as azdata from 'azdata';
+import { ConnectionModel } from './models';
+
+/**
+ * Represents a Wizard Page backed by a data model and an Azure Data Model View
+ */
+export abstract class BasePage {
```
Comment (llali) [inline generators/app/templates/ext-wizard/src/wizards/db-ops/wizard/api/basePage.ts:7]: I usually see the term "PageBase" but if this is also correct then keep it

File: generators/app/templates/ext-wizard/src/wizards/db-ops/wizard/pages/page1.ts (hunk 1 above)
Comment (llali) [reply generators/app/templates/ext-wizard/src/wizards/db-ops/wizard/pages/page1.ts:103]: space is used in existing files

---
In reply to: [465866150](https://github.com/llali/generator-azuredatastudio/pull/25#discussion_r465866150) [](ancestors = 465866150)

File: generators/app/templates/ext-wizard/src/wizards/db-ops/wizard/pages/page2.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        this.optionsDropdown = optionsDropdown;
+		return {
+			component: optionsDropdown,
+			title: 'Select a Task:'
+		};
+    }
+
+    // working with VS Code file APIs
+    private newQuery(){
+        let sqlContent = 'SELECT * FROM ' + this.model.database + '.INFORMATION_SCHEMA.TABLES';
```
Comment (llali) [inline generators/app/templates/ext-wizard/src/wizards/db-ops/wizard/pages/page2.ts:60]: I thought we're creating separate sample to show how to work with query editor.

File: generators/app/templates/ext-wizard/src/wizards/db-ops/wizard/pages/page2.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        vscode.workspace.openTextDocument({ language: 'sql', content: sqlContent }).then(async doc => {
```
Comment (llali) [inline generators/app/templates/ext-wizard/src/wizards/db-ops/wizard/pages/page2.ts:61]: >openTextDocument [](start = 25, length = 16)

does this open a sql editor with connection? I had problem doing the same thing in my extension

File: generators/app/templates/ext-wizard/src/wizards/db-ops/wizard/pages/page2.ts (hunk 4 above)
Comment (anjalia) [reply generators/app/templates/ext-wizard/src/wizards/db-ops/wizard/pages/page2.ts:60]: Ah yes sorry, this was the work done previously. I included it to get your feedback, but based on our conversation, it probably makes sense for me to remove this template.

File: generators/app/templates/ext-wizard/src/wizards/db-ops/wizard/pages/page2.ts (hunk 5 above)
Comment (anjalia) [reply generators/app/templates/ext-wizard/src/wizards/db-ops/wizard/pages/page2.ts:61]: It only opens a SQL file -- it is not automatically connected, though.

Comment (llali) [reply generators/app/templates/ext-wizard/src/wizards/db-ops/wizard/pages/page2.ts:61]: you can also connect it with azdata.queryeditor.connect(fileUri, connectionId)

---
In reply to: [465889071](https://github.com/llali/generator-azuredatastudio/pull/25#discussion_r465889071) [](ancestors = 465889071)

Comment (llali) [reply generators/app/templates/ext-wizard/src/wizards/db-ops/wizard/pages/page2.ts:61]: also using async/await is usually cleaner to write async actions

---
In reply to: [465929901](https://github.com/llali/generator-azuredatastudio/pull/25#discussion_r465929901) [](ancestors = 465929901,465889071)

**Pull request merged**